src/osd/OSD.cc

```diff
@@ -5834,8 +5834,40 @@ bool OSD::scrub_random_backoff()
   return false;
 }
 
+bool OSD::scrub_time_permit()
```

Comment:
I'd add the time arg OSD::scrub_time_permit(utime_t now) and always pass ceph_clock_now(cct) from osd code.  The unit test would pass the time values to test with.